### PR TITLE
Fix & improvements to existing vaults endpoints for apps

### PR DESCRIPTION
**Pull request by spolu**
## Description

- Small improvements (remove double check of owner, standardize the code a bit).
- Fixes the authenticator used to find the default vault in the run creation.

## Risk

Low (the only dangerous change is not used in prod as we use URLs with vaults for our own usage of Dust apps)

r? @fontanierh 

## Deploy Plan

- deploy `front`

### CODE REVIEW

File: front/pages/api/v1/w/[wId]/vaults/[vId]/apps/[aId]/runs/index.ts [hunk 1]
```diff
@@ -81,25 +81,15 @@ async function handler(
...
-  // for the legacy endpoint (not under vault, global vault assumed).
-  if (vaultId === undefined) {
-    const globalVault = await VaultResource.fetchWorkspaceGlobalVault(keyAuth);
-    vaultId = globalVault.sId;
+  // Handling the case where vId is undefined to keep support for the legacy endpoint (not under
+  // vault, global vault assumed for the auth (the authenticator associated with the app, not the
+  // user)).
+  let { vId } = req.query;
+  if (vId === undefined) {
+    vId = (await VaultResource.fetchWorkspaceGlobalVault(auth)).sId;
```
Comment (spolu) [inline front/pages/api/v1/w/[wId]/vaults/[vId]/apps/[aId]/runs/index.ts:92]: critical diff (keyAuth vs auth)

File: front/pages/api/v1/w/[wId]/vaults/[vId]/apps/[aId]/runs/[runId]/index.ts [hunk 2]
```diff
@@ -29,34 +29,24 @@ async function handler(
...
-      status_code: 404,
-      api_error: {
-        type: "workspace_not_found",
-        message: "The workspace you're trying to access was not found",
-      },
-    });
-  }
-
-  let vaultId = req.query.vId;
+  const owner = auth.getNonNullableWorkspace();
```
Comment (fontanierh) [inline front/pages/api/v1/w/[wId]/vaults/[vId]/apps/[aId]/runs/[runId]/index.ts:32]: I think we should never use non nullable workspace in handlers. This should be 404 not 500

File: front/pages/api/v1/w/[wId]/vaults/[vId]/apps/[aId]/runs/index.ts [hunk 3]
```diff
@@ -81,25 +81,15 @@ async function handler(
   auth: Authenticator,
   keyAuth: Authenticator
 ): Promise<void> {
+  const owner = auth.getNonNullableWorkspace();
```
Comment (fontanierh) [inline front/pages/api/v1/w/[wId]/vaults/[vId]/apps/[aId]/runs/index.ts:84]: same here

**Review by fontanierh — APPROVED** [review]

File: front/pages/api/v1/w/[wId]/vaults/[vId]/apps/[aId]/runs/[runId]/index.ts (hunk 2 above)
Comment (spolu) [reply front/pages/api/v1/w/[wId]/vaults/[vId]/apps/[aId]/runs/[runId]/index.ts:32]: Well here it is unexpected as we already check for this in the authentication wrapper

Comment (spolu) [reply front/pages/api/v1/w/[wId]/vaults/[vId]/apps/[aId]/runs/[runId]/index.ts:32]: We standardized on handling this in warppers and using nonnullable in most endpoints/handlers

**Pull request merged**